### PR TITLE
Add boot options to perform display rotation and set preferred screen resolution

### PR DESCRIPTION
**Pull request by martinwhitaker**
This addresses issue #381. It's not a perfect fix for that machine, due to a buggy BIOS, but hopefully it will be useful for other 2-in-1 laptops too.

As I needed to add boot command line parsing in the efisetup code, I've taken the opportunity to make the EFI debug code a run-time option instead of a compile-time option.

### CODE REVIEW

**Comment by x86fr** [comment]
I don't think I have any concerned laptop to check in real-world unfortunately, but code looks fine. 
Only potential concern is about the serial output, as it uses the framebuffer.

**Comment by martinwhitaker** [comment]
Serial output uses the text shadow buffer, which is unaffected by these changes.

**Comment by debrouxl** [comment]
It looks reasonable indeed. It's certainly safe to consider that screens which would need 180° rotation are such a cursed thing that it doesn't need to be seriously taken into account :)
I'm not entirely a fan of the partial code duplication in the two `parse_option()`, but I think I understand why it's being done that way.

File: README.md [hunk 1]
```diff
@@ -142,6 +142,16 @@ recognised:
...
       * usb
       * both
+  * screen.mode=*w*x*h* (EFI framebuffer only)
+    * where *w*x*h* is the preferred screen resolution (e.g. 1024x768)
+  * screen.mode=bios (EFI framebuffer only)
+    * uses the default screen resolution set by the UEFI BIOS
+  * screen.rhs-up (graphics mode only)
+    * rotates the display clockwise by 90 degrees
+  * screen.lhs-up (graphics mode only)
+    * rotates the display anti-clockwiseby 90 degrees
```
Comment (Nable80) [inline README.md:152]: ```s/anti-clockwiseby/counter-clockwise by/```

File: system/screen.c [hunk 2]
```diff
@@ -10,8 +10,20 @@
...
 
+#include "string.h"
+
 #include "screen.h"
 
+//------------------------------------------------------------------------------
+// Private Types
+//------------------------------------------------------------------------------
+
+typedef enum  __attribute__ ((packed)) {
```
Comment (Nable80) [inline system/screen.c:21]: ``-fshort-enums``? The ``packed`` attribute is quiet unusual for anything except structures.

File: README.md (hunk 1 above)
Comment (martinwhitaker) [reply README.md:152]: anti-clockwise is correct English.

File: system/screen.c (hunk 2 above)
Comment (martinwhitaker) [reply system/screen.c:21]: -fshort-enums would affect all enums, which we don't necessarily want.

File: README.md (hunk 1 above)
Comment (Nable80) [reply README.md:152]: My bad, I didn't come across this variant before. It looks like "anti-clockwise" is more common in en_GB, while I'm really spoiled by en_US… I'm sorry I didn't check it before commenting, thank you for clarification.

A missing space before ``by`` should be fixed anyway, shouldn't it?

File: system/screen.c (hunk 2 above)
Comment (Nable80) [reply system/screen.c:21]: This is true but it shouldn't break anything for bare-metal code.
OK, I got it, I should check it myself first.

File: README.md (hunk 1 above)
Comment (martinwhitaker) [reply README.md:152]: Yes, missing space fixed, thanks for spotting that.

**Pull request merged**